lib/Ed25519KeyPair.js

```diff
@@ -11,32 +11,73 @@ const LDKeyPair = require('./LDKeyPair');
 
 class Ed25519KeyPair extends LDKeyPair {
   /**
-   * @param options {object} See LDKeyPair constructor docstring for full list
-   *
-   * @param [options.publicKeyBase58] {string}
-   * @param [options.privateKeyBase58] {string}
+   * @classdesc  An implementation of
+   * [Ed25519 Signature 2018]{@link https://w3c-dvcg.github.io/lds-ed25519-2018/}
+   * for
+   * [jsonld-signatures.]{@link https://github.com/digitalbazaar/jsonld-signatures}
+   * @description Creates an [Ed25519]{@link https://ed25519.cr.yp.to/}
+   * Key Pair usable by a json-ld signature.
+   * @example
+   * > const ops = {
+   *   publicKeyBase58: '2S2Q6MkaFJewa',
+   *   privateKeyBase58: '8JHajxJ1pKnSxwziUGFnuzvB96psLvqejZtgcA7QKoZ2'
+   * };
+   * > const EDKey = new Ed25519KeyPair(ops);
+   * @param {KeyPairOptions} options - Keys must be in Base58
+   * other options most follow
+   * [KeyPairOptions]{@link ./index.md#KeyPairOptions}.
+   * @param {string} options.publicKeyBase58 - Public Key exposed in Signatures.
+   * @param {string} options.privateKeyBase58 - Private Key for signing.
    */
   constructor(options = {}) {
     super(options);
     this.type = 'Ed25519VerificationKey2018';
     this.privateKeyBase58 = options.privateKeyBase58;
     this.publicKeyBase58 = options.publicKeyBase58;
   }
-
+  /**
+   * @description Returns the public key.
+   * @example
+   * > edKeyPair.publicKey;
+   * 2S2Q6MkaFJewa
+   * @implements {LDKeyPair#publicKey}
+   * @readonly
+   * @returns {string} Returns the public key.
+   * @see [publicKey]{@link ./LDKeyPair.md#publicKey}
+   */
   get publicKey() {
     return this.publicKeyBase58;
   }
-
+  /**
+   * @description Returns the private key.
+   * @example
+   * > edKeyPair.privateKey;
+   * 8JHajxJ1pKnSxwziUGFnuzvB96psLvqejZtgcA7QKoZ2
+   * @implements {LDKeyPair#privateKey}
+   * @readonly
+   * @returns {string} Returns the private key.
+   * @see [privateKey]{@link ./LDKeyPair.md#privateKey}
+   */
   get privateKey() {
     return this.privateKeyBase58;
   }
 
   /**
-   * @param [options] {object} See LDKeyPair docstring for full list
-   * @param [options.seed] {(Uint8Array|Buffer)} a 32-byte seed for a
-   *   deterministic key.
+   * @description Static method that generates
+   * a KeyPair with a deterministic seed.
+   * @example
+   * > const ops = {
+   *   seed: new Int8Array(58)
```

Comment:
this code works. will try yours.

Reply:
Your code also works and show a better understanding so changed it to
new Int8Array(32)

so yeah 32 0s is fine.

Reply:
@mattcollier please review this again. I think demonstrating the deterministic seed here is a good idea. changed from bits to bytes.

Reply:
@aljones15 Int8Array and Uint8Array are not equivalent.  The API accepts a Uint8Array, which is what we need to demonstrate.  Also we want to try to demonstrate best practices.  Initializing the thing with zeros for a seed is definitely not best practice. Again, I think skipping the example for passing a seed would be fine.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Int8Array
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array

Reply:
Yeah unsigned means it can not go negative and is max 8 bits.

```
const seed = new Uint8Array(32).map(n => Math.floor(Math.random() * 255));
const options = {seed};
const key = await Ed25519KeyPair.generate(options);
```
if that is not ok then will just drop the @example for this one, issue with the above is that the seed is actually kind of random.

Reply:
Demonstrating the use of Math.random here is also not a good idea: https://stackoverflow.com/questions/5651789/is-math-random-cryptographically-secure